core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -520,6 +520,8 @@ object SparkSubmit {
       }
       if (args.principal != null) {
         require(args.keytab != null, "Keytab must be specified when the keytab is specified")
+        sysProps.put("spark.yarn.keytab", args.keytab)
+        sysProps.put("spark.yarn.principal", args.principal)
```

Comment:
We might want to look at yarn Client.scala setupCredentials() since its doing something pretty similar it looks like.